mystery-deckbuilder/Assets/Scripts/Encounter/BarScript.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+/**
+ * generic script for controling a bar object
+ */
+public class BarScript : MonoBehaviour   
+{
+    public Slider slider;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    public void Initialize(int maxVal, int val)
+    {
+        SetMax(maxVal);
+        SetValue(val);
+    }
```

Comment:
NIT: While the Set functions do not need any commenting, an argument could be made that Initialize is not trivial. In this case though, implementation is *trivial enough* to understand at a glance